File: Darts/src/burlton/dartzee/code/db/sanity/SanityCheckResultFinalScoreMismatch.java
package burlton.dartzee.code.db.sanity;

import burlton.core.code.obj.HandyArrayList;
import burlton.core.code.obj.SuperHashMap;
import burlton.dartzee.code.bean.ScrollTableDartsGame;
import burlton.dartzee.code.db.GameEntityKt;
import burlton.dartzee.code.db.ParticipantEntity;
import burlton.desktopcore.code.bean.ScrollTable;
import burlton.desktopcore.code.util.TableUtil.DefaultModel;

import javax.swing.table.DefaultTableModel;
import java.sql.Timestamp;

public final class SanityCheckResultFinalScoreMismatch extends AbstractSanityCheckResult
{
	private int gameType = -1;
	private SuperHashMap<ParticipantEntity, Integer> hmParticipantToFinalScore = new SuperHashMap<>();
	
	public SanityCheckResultFinalScoreMismatch(int gameType, SuperHashMap<ParticipantEntity, Integer> hmParticipantToFinalScore)
	{
		this.gameType = gameType;
		this.hmParticipantToFinalScore = hmParticipantToFinalScore;
	}
	
	@Override
	public String getDescription()
	{
		String gameDesc = GameEntityKt.getTypeDesc(gameType);
		return "FinalScores that don't match the raw data (" + gameDesc + ")";
	}
	
	@Override
	public ScrollTable getScrollTable()
	{
		return new ScrollTableDartsGame("GameId");
	}

	@Override
	public int getCount()
	{
		return hmParticipantToFinalScore.size();
	}

	@Override
	protected DefaultTableModel getResultsModel()
	{
		DefaultModel model = new DefaultModel();
		model.addColumn("ParticipantId");
		model.addColumn("PlayerId");
		model.addColumn("GameId");
		model.addColumn("DtLastUpdate");
		model.addColumn("FinalScore");
		model.addColumn("FinalScoreRAW");
		
		HandyArrayList<ParticipantEntity> pts = hmParticipantToFinalScore.getKeysAsVector();
		for (ParticipantEntity pt : pts)
		{
			long participantId = pt.getRowId();
			long playerId = pt.getPlayerId();
			long gameId = pt.getGameId();
			Timestamp dtLastUpdate = pt.getDtLastUpdate();
			int finalScore = pt.getFinalScore();
			int finalScoreRaw = hmParticipantToFinalScore.get(pt);
			
			Object[] row = {participantId, playerId, gameId, dtLastUpdate, finalScore, finalScoreRaw};
			model.addRow(row);
		}
		
		return model;
	}

}

File: Darts/src/burlton/dartzee/code/db/sanity/SanityCheckResultFinalScoreMismatch.kt
package burlton.dartzee.code.db.sanity

import burlton.dartzee.code.bean.ScrollTableDartsGame
import burlton.dartzee.code.db.ParticipantEntity
import burlton.dartzee.code.db.getTypeDesc
import burlton.desktopcore.code.util.TableUtil.DefaultModel
import javax.swing.table.DefaultTableModel

class SanityCheckResultFinalScoreMismatch(private val gameType: Int, private val hmParticipantToFinalScore: MutableMap<ParticipantEntity, Int>) : AbstractSanityCheckResult()
{
    override fun getDescription() = "FinalScores that don't match the raw data (${getTypeDesc(gameType)})"

    override fun getScrollTable() = ScrollTableDartsGame("GameId")

    override fun getCount() = hmParticipantToFinalScore.size

    override fun getResultsModel(): DefaultTableModel
    {
        val model = DefaultModel()
        model.addColumn("ParticipantId")
        model.addColumn("PlayerId")
        model.addColumn("GameId")
        model.addColumn("DtLastUpdate")
        model.addColumn("FinalScore")
        model.addColumn("FinalScoreRAW")

        val pts = hmParticipantToFinalScore.keys
        for (pt in pts)
        {
            val participantId = pt.rowId
            val playerId = pt.playerId
            val gameId = pt.gameId
            val dtLastUpdate = pt.dtLastUpdate
            val finalScore = pt.finalScore
            val finalScoreRaw = hmParticipantToFinalScore[pt]!!

            val row = arrayOf(participantId, playerId, gameId, dtLastUpdate, finalScore, finalScoreRaw)
            model.addRow(row)
        }

        return model
    }

}
